runtime/appruntime/metrics/datadog_exporter.go

```diff
@@ -0,0 +1,50 @@
+package metrics
+
+import (
+	"os"
+
+	"encore.dev/appruntime/config"
+	"encore.dev/appruntime/metadata"
+	"encore.dev/appruntime/metrics/datadog"
+)
+
+func init() {
+	registerProvider(providerDesc{
+		name: "datadog",
+		matches: func(cfg *config.Metrics) bool {
+			return cfg.Datadog != nil
+		},
+		newExporter: func(m *Manager) exporter {
+			containerMetadata, err := metadata.GetContainerMetadata(m.cfg.Runtime)
+			if err != nil {
+				m.rootLogger.Err(err).Msg("unable to initialize metrics exporter: error getting container metadata")
+				return nil
+			}
+
+			cfg := m.cfg.Runtime.Metrics.Datadog
+			if cfg.Site == "" {
+				m.rootLogger.Err(err).Msg("unable to initialize metrics exporter: Datadog site unset")
+				return nil
+			}
+
+			err = os.Setenv("DD_SITE", cfg.Site)
```

Comment:
Is this necessary? Is there no other way of getting the site information and API key to the client? It makes it easy for rogue libraries to sniff credentials. If the libraries are designed that way I guess there's not too much we can do about it, but would be nice to avoid.